docs/tutorials/upgrade-extension.md

```diff
@@ -100,15 +88,15 @@ spec:
         metadata:
           annotations:
             kubectl.kubernetes.io/last-applied-configuration: |
-              {"apiVersion":"olm.operatorframework.io/v1","kind":"ClusterExtension","metadata":{"annotations":{},"name":"argocd"},"spec":{"namespace":"argocd","serviceAccount":{"name":"argocd-installer"},"source":{"catalog":{"packageName":"argocd-operator","version":"0.5.0"},"sourceType":"Catalog"}}}
-          creationTimestamp: "2024-11-11T14:13:12Z"
+              {"apiVersion":"olm.operatorframework.io/v1","kind":"ClusterExtension","metadata":{"annotations":{},"name":"argocd"},"spec":{"namespace":"argocd","serviceAccount":{"name":"argocd-installer"},"source":{"catalog":{"packageName":"argocd-operator","version":"0.2.1"},"sourceType":"Catalog"}}}
```

Comment:
I have a feeling that it should be `0.2.0` if we are upgrading from `0.2.0` to `0.2.1`.

I think it would be really good to have a reproducible example here. Something people can run end to end and see how it works.

Reply:
> Should we change the ClusterExtension sample argocd manifest to match our docs at version 0.2.0 or 0.2.1? Or is it good enough to make these docs examples just work and leave the manifest alone at 0.6.0?

I think the above comment is related to what you are asking here.

On one hand - having a asample which installs 0.2.0 version, while there is 0.12.0 already, is not great. But on the other hand - if it helps us to demonstrate things in docs, the it might be worth doing it.

Alternatively, we can include all the manifests (RBAC, service account, etc) required for 0.2.0 & 0.2.1 just in docs and make docs less reliant on the sample manifest. I think doing this might be the best approach here.

Reply:
That's an interesting idea. Do you think it would make the most sense to have those manifests somewhere in the beginning of this "getting started" doc?

And as for the output version here on line 91, it's showing 0.2.1 because this is the verification output after you perform the upgrade.

Reply:
>I expected the spec to have a new version, but kubectl.kubernetes.io/last-applied-configuration to have the previous spec (before we applied a new version). Is it not how it should work?

That annotation matches my expectations. We patched the ClusterExtension to 0.2.1 in using kubectl so it updated this annotation as part of that patch since we are applying a "new" configuration.

Reply:
I'm +1 for a collapsible admonition if we are going to deviate from the sample.

Reply:
I can make a note admonition explaining this nuance with the last-applied-configuration value

Reply:
You are right. I was patching. All good then. Resolving this thread.